controllers/subjectpermission/subjectpermission_controller.go

```diff
@@ -152,6 +152,16 @@ func (r *SubjectPermissionReconciler) Reconcile(ctx context.Context, request ctr
 		return ctrl.Result{}, err
 	}
 
+	// eliminate non terminating and non existing Namespace from the nsList.Items
+	newNsList := corev1.NamespaceList{}
+	for i := range nsList.Items {
+		if controllerutil.ValidateNamespace(&nsList.Items[i]) {
+			newNsList.Items = append(newNsList.Items, nsList.Items[i])
+		} else {
+			reqLogger.Info(fmt.Sprintf("Namespace %s doesn't exist or in terminating state", nsList.Items[i].Name))
```

Comment:
A minor query here, but I see that `ValidateNamespace` will treat a namespace as invalid if its `Name` is empty; it would then attempt to be logged here. So it could show up strange as a log message. Maybe just surrounding it with quotes will be enough.

```suggestion
			reqLogger.Info(fmt.Sprintf("Namespace '%s' doesn't exist or in terminating state", nsList.Items[i].Name))
```

Reply:
Thanks for the review @mrbarge
I have addressed all the comments in my latest commit.